### ISSUE TITLE
ReadTimeout errors not being caught and/or logged

### ISSUE THREAD

**Issue by luisobo** [mongodb-labs/elastic2-doc-manager#8]
We experienced some connectivity issues with our ES cluster and the reads from ES where timing out. It retried a few times but it ended up giving up.

The problem is that those read timeout errors where not logged into the logger and we were not alerted of this fact, they ended up in stderr, I believe, and not in the log file that we configured in mongo-connector's `config.json`.

Here is the trace
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.4/site-packages/mongo_connector-2.3-py3.4.egg/mongo_connector/util.py", line 85, in wrapped
    func(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/mongo_connector-2.3-py3.4.egg/mongo_connector/oplog_manager.py", line 304, in run
    ns, timestamp)
  File "/usr/local/lib/python3.4/site-packages/mongo_connector-2.3-py3.4.egg/mongo_connector/util.py", line 32, in wrapped
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/elastic2_doc_manager-0.2.0-py3.4.egg/mongo_connector/doc_managers/elastic2_doc_manager.py", line 164, in update
    self.upsert(updated, namespace, timestamp)
  File "/usr/local/lib/python3.4/site-packages/mongo_connector-2.3-py3.4.egg/mongo_connector/util.py", line 32, in wrapped
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/elastic2_doc_manager-0.2.0-py3.4.egg/mongo_connector/doc_managers/elastic2_doc_manager.py", line 181, in upsert
    refresh=(self.auto_commit_interval == 0))
  File "/usr/local/lib/python3.4/site-packages/elasticsearch-2.3.0-py3.4.egg/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/elasticsearch-2.3.0-py3.4.egg/elasticsearch/client/__init__.py", line 279, in index
    _make_path(index, doc_type, id), params=params, body=body)
  File "/usr/local/lib/python3.4/site-packages/elasticsearch-2.3.0-py3.4.egg/elasticsearch/transport.py", line 329, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/usr/local/lib/python3.4/site-packages/elasticsearch-2.3.0-py3.4.egg/elasticsearch/connection/http_urllib3.py", line 102, in perform_request
    raise ConnectionTimeout('TIMEOUT', str(e), e)
elasticsearch.exceptions.ConnectionTimeout: ConnectionTimeout caused by - ReadTimeoutError(HTTPSConnectionPool(host='XXXXX', port=9243): Read timed out. (read timeout=100))
```

**Comment by llvtt** [comment]
What version of mongo-connector are you using? This looks like an issue that was fixed in [this commit](https://github.com/pdspicer/mongo-connector/commit/f2fce424d8135ea96724301da0a117343500f80a) and was released as part of version 2.4.

**Comment by luisobo** [comment]
@llvtt Definitely < 2.4. Will upgrade, thanks.
